pkg/destroy/clusterapi/destroy.go

```diff
@@ -0,0 +1,36 @@
+package clusterapi
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/cmd/openshift-install/command"
+	"github.com/openshift/installer/pkg/clusterapi"
+)
+
+// DestroyClusterAPIResources removes all cluster api artifacts that are not destroyed by
+// other destroy processes.
+func DestroyClusterAPIResources() error {
+	artifacts := []string{
+		// The Artifacts directory does not contain assets. These assets would be
+		// removed with a Purge and the directory would be automatically removed when all assets are gone.
+		// Remove the Artifacts Directory on cluster destroy
+		filepath.Join(command.RootOpts.Dir, clusterapi.ArtifactsDir),
+	}
+
+	for _, artifact := range artifacts {
+		if _, err := os.Stat(artifact); err == nil {
+			if !os.IsNotExist(err) {
```

Comment:
You only enter here if `err == nil` so this check doesn't make sense. Also, we should use now `errors.Is(err, fs.ErrNotExist)` instead.